src/clr-angular/utils/focus/focusable-item/linkers.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Observable } from 'rxjs';
+import { Direction } from '../direction.enum';
+import { FocusableItem } from './focusable-item';
+
+/**
+ * Links a set of focusable items to a parent along one direction
+ */
+export function linkParent(
+  items: FocusableItem[],
+  parent: FocusableItem | Observable<FocusableItem>,
+  direction: Direction
+) {
+  items.forEach(item => (item[direction] = parent));
+}
+
+/**
+ * Double-links a set of focusable items vertically, possibly looping
+ */
+export function linkVertical(items: FocusableItem[], loop = true) {
+  items.forEach((item, index) => {
+    if (index > 0) {
+      item.up = items[index - 1];
+    }
+    if (index < items.length - 1) {
+      item.down = items[index + 1];
+    }
+  });
+  if (loop && items.length) {
```

Comment:
Shouldn't this guard another condition where `items.length > 1`? Otherwise, it would link to itself?

Reply:
`if (items.length)` and `if (items.length > 1)` do the same thing, 0 is the only falsy number in Javascript.